pybaseball/datahelpers/caching/__init__.py

```diff
@@ -0,0 +1,117 @@
+        self.cache_config_override = cache_config_override
+        self.reset_cache = reset_cache
+        self.reset_index = reset_index
+        _mkdir(self.cache_config.cache_directory)
+
+    def __call__(self, func: Callable[..., pd.DataFrame]) -> Callable[..., _ProfiledCachedDataFrame]:
+        def _cached(*args: Any, **kwargs: Any) -> _ProfiledCachedDataFrame:
+            # Skip all this if caching is disabled
+            if not self.cache_config.enabled:
+                return func(*args, **kwargs)
+
+            read_time = None
+            write_time = None
+            filename = None
+
+            try:
+                f_hash = get_func_hash(func, args, kwargs)
+                filename = os.path.join(self.cache_config.cache_directory, f"{f_hash}.{self.extension}")
+
+                if os.path.exists(filename):
+                    modified = datetime.fromtimestamp(os.path.getmtime(filename))
+                    if (modified + self.cache_config.expiration) > datetime.now():
+                        # Hasn't expired yet
+                        if self.cache_config.profiling:
+                            start = time.perf_counter()
+                            data = self.load(filename)
+                            read_time = time.perf_counter() - start
+                        else:
+                            data = self.load(filename)
+                        if self.reset_index:
+                            data = data.reset_index(drop=True)
+                        if read_time:
+                            return data, read_time, write_time, os.path.getsize(filename)
+                        else:
+                            return data
+            except:  # pylint: disable=broad-except
+                # If this fails, log it, and then go ahead and make the function call
+                # No need to crash the real work in this phase
+                for ex in traceback.format_exc().split('\n'):
+                    logging.debug(ex)
+            result = func(*args, **kwargs)
+
+            if filename and result is not None and isinstance(result, pd.DataFrame) and not result.empty:
+                if self.reset_index:
+                    result = result.reset_index(drop=True)
+                if self.cache_config.profiling:
+                    start = time.perf_counter()
+                    self.save(result, str(filename))
+                    write_time = time.perf_counter() - start
+                else:
+                    self.save(result, filename)
+
+            if write_time and filename:
+                return result, read_time, write_time, os.path.getsize(filename)
+            else:
+                return result
+
+        if self.reset_cache:
+            self.bust_cache(func)
+
+        return _cached
+
+    def bust_cache(self, func: Callable) -> None:
+        cache_files = [
+            x for x in os.walk(self.cache_config.cache_directory).__next__()[2] if x.startswith(f"{func.__name__}(")
```

Comment:
????

Reply:
@schorrm this comment is not helpful.

Reply:
Walk next 2? It's unclear what this does / how?

Reply:
Okay, I'll break this down a bit to be clearer about what it is doing.

Reply:
Actually since this should be flat I'll switch this to `os.listdir`

Reply:
Done